ospfd/ospf_lsa.c

```diff
@@ -167,6 +167,16 @@ struct ospf_lsa *ospf_lsa_new()
 	return new;
 }
 
+struct ospf_lsa *ospf_lsa_new_and_data(size_t size)
+{
+	struct ospf_lsa *new;
+
+	new = ospf_lsa_new();
+	new->data = ospf_lsa_data_new(size);
```

Comment:
May fail if ospf_lsa_new() return null.
Need to check that new is not null before calling ospf_lsa_data_new()